yagm.go

```diff
@@ -94,17 +98,31 @@ func (mux *YagmMux) HandleFunc(pattern string, handler func(http.ResponseWriter,
 func (mux *YagmMux) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	var handler http.Handler
 
-	for _, route := range mux.routes {
-		if route.match(r.URL.Path) {
-			handler = route.handler
+    hash := sha1.New()
```

Comment:
That gives me this error for some reason: `.\yagm.go:104: r.URL.path undefined (type *url.URL has no field or method path, but does have Path)`